packages/modal/components/DialogModal.tsx

```diff
@@ -20,6 +24,12 @@ export interface DialogModalProps extends ModalBaseProps {
   isContentFlush?: boolean;
   /** The text displayed in the header of the modal. */
   title: React.ReactNode;
+  /** Controls the icon type displayed next to the title. */
+  iconShape?: IconShapes | React.ReactElement<HTMLElement>;
+  /** Icon color can be customized otherwise it will inherit color. */
+  iconColor?: React.CSSProperties["color"];
+  /** Icon size can be customized, it is defaulted to "xs" */
+  iconSize?: IconSize;
```

Comment:
💬 What do you think about converting this to an object like:

icon: {
  shape: ...;
  color: ...;
  size: ...;
}
I'm not to opinionated about this, but it instantly came to my mind after seeing the repeated icon prefix.

Reply:
I like Max's idea too. Makes it more extensible.